src/util/dev/server.ts

```diff
@@ -329,11 +347,70 @@ export default class DevServer {
       }
     }
 
+    const apiFiles = await getApiFiles(this.cwd, this.output);
+
+    if (apiFiles.length > 0) {
+      if (!config.builds || config.builds.length === 0) {
+        config.builds = [];
+
+        if (pkg) {
+          const staticBuilder = await detectBuilder(pkg);
+
+          if (staticBuilder) {
+            config.builds.push(staticBuilder);
+          }
+        }
+
+        if (apiFiles && apiFiles.length > 0) {
```

Comment:
I don't think this check is necessary since `if (apiFiles.length > 0) {` is already being checked on line 352.